http-client/Network/HTTP/Client/Request.hs

```diff
@@ -370,26 +370,32 @@ requestBuilder req Connection {..} = do
     toTriple (RequestBodyStream len stream) = do
         -- See https://github.com/snoyberg/http-client/issues/74 for usage
         -- of flush here.
-        let body = writeStream False stream
+        let body = writeStream False (fromIntegral len) stream
             -- Don't check for a bad send on the headers themselves.
             -- Ideally, we'd do the same thing for the other request body
             -- types, but it would also introduce a performance hit since
             -- we couldn't merge request headers and bodies together.
             now  = flushHeaders (Just len) >> checkBadSend body
         return (Just len, now, body)
     toTriple (RequestBodyStreamChunked stream) = do
-        let body = writeStream True stream
+        let body = writeStream True 0 stream
             now  = flushHeaders Nothing >> checkBadSend body
         return (Nothing, now, body)
     toTriple (RequestBodyIO mbody) = mbody >>= toTriple
 
-    writeStream isChunked withStream =
-        withStream loop
+    writeStream isChunked len withStream =
```

Comment:
The `isChunked` and `len` is really providing redundant information. How about having a `mlen :: Maybe Int` parameter?